.github/workflows/on-push.yml

```diff
@@ -21,50 +34,56 @@ jobs:
           npx lerna run --scope=lean4 package
       - name: Upload artifact
         uses: actions/upload-artifact@v2
+        if: matrix.os == 'ubuntu-latest'
         with:
           name: vscode-lean4
           path: 'vscode-lean4/lean4-*.vsix'
       - name: Lint
         run: npm run lint
-
-      - name: Download Lean4 build artifact
-        uses: lovettchris/action-download-artifact@v2.15.11
-        with:
-          name: build-Linux release
-          path: .
-          github_token: ${{secrets.GITHUB_TOKEN}}
-          workflow: ci.yml
-          workflow_conclusion: success
-          branch: master
-          repo: leanprover/lean4
-          check_artifacts:  true
-      - name: Unpack Lean4 build
+      - name: Install Brew Packages
         run: |
-          mkdir lean4
-          tar -xvf lean-*-linux.tar.zst -C lean4 --strip-components 1
-          ls -al lean4/bin
-          echo ================= ldd lean
-          ldd lean4/bin/lean
-          echo ================= ldd libleanshared.so
-          ldd lean4/lib/lean/libleanshared.so
-          chmod u+x lean4/bin/*
-      - name: Set path to elan
+          brew install ccache tree zstd coreutils
+        if: matrix.os == 'macos-latest'
+      - name: Set path to elan on Linux or macOS
+        if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
         run: |
           echo "$HOME/.elan/bin" >> $GITHUB_PATH
-      - name: Setup elan toolchain
+      - name: Setup elan toolchain on Linux or macOS
+        if: matrix.os == 'ubuntu-latest' || matrix.os == 'macos-latest'
         run: |
           curl -O --location https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh
           chmod u+x elan-init.sh
           ./elan-init.sh -y --default-toolchain leanprover/lean4:nightly
           echo Checking location "$HOME/.elan/bin"...
-          ls -al "$HOME/.elan/bin"
+          ls -al $HOME/.elan/bin
           elan --version
-          elan toolchain link master "$(pwd)/lean4"
+          mkdir temp
+          cp -r $HOME/.elan//toolchains/leanprover--lean4---nightly/* temp
+          elan toolchain link master temp
+
+      - name: Set path to elan on Windows
+        shell: pwsh
+        if: matrix.os == 'windows-latest'
+        run: |
+          echo "$HOME\.elan\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```

Comment:
The $GITHUB_PATH doesn't work on windows?

Reply:
This is appending to the GITHUB_PATH file, it's just in powershell you access an environment variable with the `$env:` prefix.  I probably can do `echo `$HOME\.elan\bin" >> $env:GITHUB_PATH` here.

Reply:
Ah, makes sense.